src/dag/get.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const dagPB = require('ipld-dag-pb')
+const dagCBOR = require('ipld-dag-cbor')
+const promisify = require('promisify-es6')
+const CID = require('cids')
+const waterfall = require('async/waterfall')
+const block = require('../block')
+
+module.exports = (send) => {
+  return promisify((cid, path, options, callback) => {
+    if (typeof path === 'function') {
+      callback = path
+      path = undefined
+    }
+
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+
+    options = options || {}
+
+    if (CID.isCID(cid)) {
+      cid = cid.toBaseEncodedString()
+    }
+
+    if (typeof cid === 'string') {
+      const split = cid.split('/')
+      cid = split[0]
+      split.shift()
+
+      if (split.length > 0) {
+        path = split.join('/')
+      } else {
+        path = '/'
+      }
+    }
+
+    waterfall([
+      cb => {
+        send({
+          path: 'dag/resolve',
+          args: cid + '/' + path,
+          qs: options
+        }, cb)
+      },
+      (resolved, cb) => {
+        block(send).get(new CID(resolved['Cid']['/']), (err, blk) => cb(err, blk, resolved['RemPath']))
+      },
+      (blk, path, cb) => {
+        if (blk.cid.codec === 'dag-cbor') {
+          dagCBOR.resolver.resolve(blk, path, cb)
+        }
+        if (blk.cid.codec === 'dag-pb') {
+          dagPB.resolver.resolve(blk, path, cb)
+        }
+      }
+    ], callback)
```

Comment:
@magik6k this is an interesting approach. It misses any other format that is not `dag-cbor` and or `dag-pb`. I also believe that it won't pass every test currently present in https://github.com/ipfs/interface-ipfs-core/blob/master/src/dag.js, which is a requisite to get things merged in js-ipfs-api.

I had understood that you were working on implementing the dag api on go-ipfs, is there anything blocking you there from finishing the implementation?